### ISSUE TITLE
extract months

### ISSUE THREAD

**Issue by tito-am** [ecor/RMAWGEN#2]
Hi,

I have the following error:

Error in extractmonths(data = 1:nrow(generation00$prec_gen), when = c("Dec",  : 
  could not find function "extractmonths"

I'm following the precipitation-generator.R example and after installing from source (windows zip) and from repositories (CRAN) i have the same error. I also tried sessionInfo() to be sure the package is loaded. extractdays is identified but it's not the case for extractmonths.

thank you!

Tito

**Comment by ecor** [comment]
Hi. This happens because 'lubridate' ("tidyverse") package masks date object, then  'months' methods  within the function works in a different way as expected.  Thank you for the message, I'm trying to fix it. Emanuele

**Comment by ecor** [comment]
Are RMAWGEN loaded in your session?  Function should be found. Take care that 'extractmonths' may not work if 'lubridate' is also loaded because of a new definition of date object masking the one of 'base' package. Thank you Emanuele

**Comment by tito-am** [comment]
Hi Emmanuele,

Thanks for your quick response. I sourced extractmonths and it solved the
problem. Lubridate was creating problems as well but I changed for zoo as
well.

I'm currently trying the generator with my own data. And when I try
*ComprehensivePrecipitationGenerator
i get:*

Error in spline(x = x, y = val[, i], xout = x_out) : zero non-NA points

I'm beggining slow and I'm trying to generate precipitation for one
station. Maybe that's the problem.

Thank you for your help!


Le jeu. 13 juin 2019 à 04:45, Emanuele Cordano <notifications@github.com> a
écrit :
